doc/Reference/Markup/Binding101.md

```diff
@@ -0,0 +1,174 @@
+
+```cs
+public partial class MainPage : Page
+{
+	public MainPage()
+	{
+		this.DataContext<MyViewModel>((page, vm) => page
+			.Content(new TextBlock()
+				.Text(() => vm.Message)
+			);
+	}
+}
+```
+
+Alternatively we may want to reduce the overhead on the binding by only having this fire once to set the Text Property and not be watched for changes for example if we're setting a title or something that doesn't change. In this case we can use the `Mode` method to set the binding mode to `OneTime`.
+
+```cs
+public partial class MainPage : Page
+{
+	public MainPage()
+	{
+		this.DataContext<MyViewModel>((page, vm) => page
+			.Content(new TextBlock()
+				.Text(x => x.Bind(() => vm.Title)
+					.Mode(BindingMode.OneTime))
+			));
+	}
+}
+```
+
+The BindingMode additionally provides you with some shorthand methods to set the BindingMode to `OneWay`, `TwoWay` or `OneTime`.
+
+```cs
+public partial class MainPage : Page
+{
+	public MainPage()
+	{
+		this.DataContext<MyViewModel>((page, vm) => page
+			.Content(new TextBlock()
+				.Text(x => x.Bind(() => vm.Title)
+					.OneTime())
+			));
+	}
+}
+```
+
+Sometimes you may want to update the format of the value before it is displayed. In this case you can use the `Convert` method to provide a converter to the binding.
+
+```cs
+public partial class MainPage : Page
+{
+	public MainPage()
+	{
+		this.DataContext<MyViewModel>((page, vm) => page
+			.Content(new TextBlock()
+				.Text(x => x.Bind(() => vm.Query)
+					.Convert(query => $"Search: {query}"))
+			));
+	}
+}
+```
+
+You can also use the shorthand version of this to simply provide the binding and a converter.
+
+```cs
+public partial class MainPage : Page
+{
+	public MainPage()
+	{
+		this.DataContext<MyViewModel>((page, vm) => page
+			.Content(new TextBlock()
+				.Text(() => vm.Query, query => $"Search: {query}")
+			));
+	}
+}
+```
+
+Similarly you may need to convert the value back to the original type when the value is updated. In this case you can use the `ConvertBack` method to provide a converter to the binding.
+
+```cs
+public partial class MainPage : Page
+{
+	public MainPage()
+	{
+		this.DataContext<MyViewModel>((page, vm) => page
+			.Content(new TextBox()
+				.Text(x => x.Bind(() => vm.Enabled)
+					.Convert(enabled => $"Enabled: {enabled}")
+					.ConvertBack(text => bool.TryParse(text.Replace("Search: ", ""), out var enabled) ? enabled : false))
+			));
+	}
+}
+```
+
+## Binding to other sources
+
+Sometimes you may want or need to bind to your ViewModel. The first case we'll take a look at is one in which we want to bind to a property of another element. In this case we will use the ElementName method on the BindingFactory to update our binding so that it will bind to a named element in the Visual Tree.
```

Comment:
```suggestion
Sometimes you may want or need to bind to your ViewModel. The first case we'll take a look at is one in which we want to bind to a property of another element. In this case we will use the `ElementName` method on the `BindingFactory` to update our binding so that it will bind to a named element in the Visual Tree.
```